src/MuTalk-Utilities/Dictionary.extension.st

```diff
@@ -0,0 +1,22 @@
+Extension { #name : 'Dictionary' }
+
+{ #category : '*MuTalk-Utilities' }
+Dictionary >> addAllWithoutOverwrite: aKeyedCollection [
+
+	aKeyedCollection == self ifFalse: [
+		aKeyedCollection associationsDo: [ :assoc |
+			self addWithoutOverwrite: assoc ] ].
+	^ aKeyedCollection
+]
+
+{ #category : '*MuTalk-Utilities' }
+Dictionary >> addWithoutOverwrite: anAssociation [
+
+	| index element |
+	index := self findElementOrNil: anAssociation key.
+	element := array at: index.
+	element
+		ifNil: [ self atNewIndex: index put: anAssociation ]
+		ifNotNil: [ element value: element value + anAssociation value ].
+	^ anAssociation
+]
```

Comment:
Could we avoid having extension methods for this?
The first method is just an iteration on the argument, it could be moved to the client.
The second method could be replaced by an at:ifPresent:ifAbsent: or something like that right?

Adding an extension here could lead to problems because
- we are not relying on public APIs (that may change)
- adds a method to all dictionaries, but that method is pretty specific to dictionaries with integers as values

Reply:
I see, I didn't notice there was a at:ifPresent:ifAbsentPut: method. I'll try to change that

Reply:
Thanks @DurieuxPol !